### DISCUSSION TITLE
New Architecture native errors reporting

### DISCUSSION THREAD

**Discussion by krystofwoldrich** [reactwg/react-native-new-architecture#109] [Q&A]
As discussed in the issue https://github.com/facebook/react-native/issues/34923 (based on @getsentry adding support for the New Architecture https://github.com/getsentry/sentry-react-native/issues/2226#issuecomment-1278792357), the new architecture native errors are quite unreadable with exception of iOS Turbo Modules.

The issue https://github.com/facebook/react-native/issues/34923 describes Java RuntimeException that ends up as a Cpp crash without any information pointing toward the origin, toward Java.

Pure Cpp Modules, like the example added in RN 0.71 https://github.com/facebook/react-native/commit/d07575b1c63c214f74fd72b329892c7cfde5df69 are in a very similar situation as the Java exception. With the difference that the error is propagated to JS, but also without any traces of the original error.  I see that from the error name there should be at least the function name, but in my case, it ends up `<unknown>`.

Example of the Cpp crash. Error `Exception in HostFunction: <unknown>`
![Screenshot 2023-01-30 at 10 09 50](https://user-images.githubusercontent.com/31292499/215778110-1ee7440c-c31c-42f4-afa4-9d26740a28ad.png)

I personally like that the error is propagated back to the JS layer, I think it's very convenient for RN developers. I could imagine the original stack trace could be saved in the JS error in the [`error.cause`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause)property.

I would like to discuss here, how this could be made better, what the solution could look like, what can be achieved etc...

**Comment by cortinico** [comment]
> I would like to discuss here, how this could be made better, what the solution could look like, what can be achieved etc...

Thanks for raising this point @krystofwoldrich 

Error reporting is a known issue in the New Architecture space. As you said, ideally having the stacktrace propagated to JS would be beneficial. We're open to contribution or collaboration on this space. Specifically if folks at Sentry are interested in contributing on this part fo the New Architecture experience, we'd more than happy to support it (cc @marandaneto)

**Reply by marandaneto** [reply]
We're currently discussing this internally, we're gonna follow up on that.
